pkg/services/baremetal/host/host.go

```diff
@@ -1188,7 +1193,12 @@ EOF_POST_INSTALL_SCRIPT
 		record.Warnf(s.scope.HetznerBareMetalHost, "ExecuteInstallImageFailed", out.StdOut)
 		return actionError{err: fmt.Errorf("failed to execute installimage: %w", out.Err)}
 	}
-
+	if !strings.Contains(out.StdOut, postInstallScriptFinished) {
+		record.Warnf(s.scope.HetznerBareMetalHost, "ExecuteInstallImageFailed", out.StdOut)
+		return actionError{err: fmt.Errorf("did not find marker %q in stdout. Failed to execute installimage: %w",
+			postInstallScriptFinished,
+			out.Err)}
```

Comment:
out. Err is nil at this point. This will create a nil pointer error afaik with fmt.Errorf

Reply:
@janiskemper thank you for your sharp eyes! Yes, the err-case is handled a line above with "return".